pkg/certs/rotate.go

```diff
@@ -0,0 +1,219 @@
+package certs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/pkg/config"
+	"github.com/loft-sh/vcluster/pkg/constants"
+	"github.com/loft-sh/vcluster/pkg/util/servicecidr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Info struct {
+	Filename   string    `json:"filename,omitempty"`
+	Subject    string    `json:"subject,omitempty"`
+	Issuer     string    `json:"issuer,omitempty"`
+	ExpiryTime time.Time `json:"expiryTime"`
+	Status     string    `json:"status,omitempty"` // "OK", "EXPIRED"
+}
+
+// Rotate rotates the certificates in the PKI directory.
+// If running non-standalone it also updates the cert secret to contain the newly created certificates.
+// Depending on the withCA argument this either means rotation the leaf certificates (withCA=false)
+// or the whole PKI infra (withCA=true). In both cases the current SA pub and private key are untouched.
+// IMPORTANT NOTE: The function expects the clients in vConfig to be populated.
```

Comment:
Rather than just noting this in a comment, maybe check for the fields and do the [setup](https://github.com/loft-sh/vcluster/pull/2976/files?w=1#diff-9fee33c88e62e8cce171f5f22d07b3d49f773c07df181c833ab7727285225b86R54-R67) here if they're missing, or just always do the setup here instead of passing in the `vConfig`?

Reply:
Makes sense. Though, this requires a bit more restructuring because this would currently introduce an import cycle.  I would need `setup` here while `setup` already imports `certs`.